Launchpad: focused launchpad on My Home sends specific `context` prop to tracks events

p-jackson opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to launchpad tracks audit pfYzsZ-1KX-p2#comment-1696

## Proposed Changes

* Focused launchpad uses `focused-customer-home` as the `context` param for tracks events and API calls
  * The My Home card version of the launchpad still uses `customer-home`
  * The legacy stepper version of the launchpad still uses `onboarding` 
* Previously the launchpad context value depended on A/B test assignment, and a `null` context value was used to pause loading the launchpad until the experiment assignment had loaded. This code has been removed.
* Since I was in the area, I generalised `useLaunchpadDismisser()` so that it wasn't hardcoded to use `context=customer-home`.
  * This doesn't change any behaviour (`useLaunchpadDismisser()` was only used when context was `customer-home`)

The reason it's safe to remove the `customer-home-treatment-cumulative` context value is that this Jetpack PR has already been deployed, which removes the only use of this context value. https://github.com/Automattic/jetpack/pull/42193

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

Related to launchpad tracks audit pfYzsZ-1KX-p2#comment-1696

Launch tracks events sent from the focused launchpad will now be distinguishable from those send by the My Home card.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Test the legacy stepper launchpad
  * Use a signup flow like `/setup/newsletter`
  * When you land on the full screen stepper launchpad confirm the `context` prop for launchpad tracks events is still `onboarding`
* Test the focused launchpad
  * Create a site with `/setup/onboarding`
  * When you land on the focused My Home launchpad confirm the `context` prop for launchpad tracks events is now `focused-customer-home`
  * Complete or dismiss the focused launchpad
  * When you see the My Home card launchpad, confirm the `context` prop for launchpad tracks events is still `customer-home`

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

p-jackson committed:
Use focused-customer-home for launchpad tracks context prop

p-jackson committed:
Remove code which delayed launchpad loading till after experiment assignment

p-jackson committed:
Generalise useLaunchpadDismisser to work in multiple contexts

daledupreez commented on packages/data-stores/src/queries/use-launchpad.ts:
```diff
@@ -231,7 +235,7 @@ export const useLaunchpadDismisser = ( siteSlug: SiteSlug, checklistSlug: string
 			recordTracksEvent( 'calypso_launchpad_dismiss_guide', {
 				checklist_slug: checklistSlug,
-				context: 'customer-home',
+				context: launchpadContext,
```
Excellent catch!

daledupreez reviewed (commented):
Based on inspection this looks good to me, but it would be good to get a review and more testing from someone who's been working on the focused launchpad.

zaguiini commented on client/my-sites/customer-home/cards/launchpad/use-launchpad.ts:
```diff
@@ -12,24 +12,20 @@ import type { AppState } from 'calypso/types';
 interface UseLaunchpadProps {
 	checklistSlug: string;
-	launchpadContext: string | null;
+	launchpadContext: string;
```
Nice! I also wonder if we can rename this `useLaunchpad` hook because, at first glance, I thought we were changing the one from `packages/data-stores`. Maybe `useContextualLaunchpad` would be a better name?

p-jackson replied:
They both accept "context" params so `useContextualLaunchpad()` sort of throws me off.

But I have found it confusing to have both.

Kinda ugly, but I'm thinking maybe `useMyHomeCardLaunchpad()`? Highlights that it's a special hook that includes the collection of things that the My Home card happens to be interested in.

zaguiini replied:
Sounds good to me!

zaguiini reviewed (approved):
Works perfectly 👍🏻

p-jackson committed:
Rename hook to useMyHomeCardLaunchpad()

There's already a useLaunchpad() hook in `@automattic/data-store`, so
having two hooks with the same name is kinda confusing.
